linux_os/guide/system/accounts/accounts-pam/set_password_hashing_algorithm/set_password_hashing_algorithm_systemauth/bash/shared.sh

```diff
@@ -1,14 +1,36 @@
 # platform = multi_platform_all
 
-{{% if product in ["sle15", "sle12"] -%}}
-{{{ bash_ensure_pam_module_configuration('/etc/pam.d/common-password', 'password', 'required', 'pam_unix.so', 'sha512', '', '') }}}
-{{% elif 'ubuntu' in product -%}}
+{{{ bash_instantiate_variables("var_password_hashing_algorithm_pam") }}}
+
+{{% if 'sle' in product or 'ubuntu' in product -%}}
+PAM_FILE_PATH="/etc/pam.d/common-password"
+{{%- else -%}}
+PAM_FILE_PATH="/etc/pam.d/system-auth"
+{{%- endif %}}
+
+{{% if 'sle' in product -%}}
+CONTROL="required"
+{{%- else -%}}
```

Comment:
it appears that CONTROL will be unused if ubuntu in product

Reply:
Yes. I saw this in CI tests some minutes after filing the PR. It is not a big deal, but I will fix this.

Reply:
Fixed.